Dockerfile.ci

```diff
@@ -1393,11 +1393,10 @@ ENV AIRFLOW_REPO=${AIRFLOW_REPO}\
 RUN echo "Airflow version: ${AIRFLOW_VERSION}"
 
 # Those are additional constraints that are needed for some extras but we do not want to
-# force them on the main Airflow package.
-# Recent aiobotocore (2.6.0+) requires this specific botocore version, while
-# boto3 1.28.17 is the last version supporting that, so we pin them to help pip
-# avoid extra work. Revisit this when a new aiobotocore version is released.
-ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS="botocore==1.31.17 boto3<=1.28.17"
+# force them on the main Airflow package. Currently we need no extra limits as PIP 23.1+ has much better
+# dependency resolution and we do not need to limit the versions of the dependencies
+# aiobotocore is limited temporarily until it stops backtracking pip
+ARG EAGER_UPGRADE_ADDITIONAL_REQUIREMENTS=""
```

Comment:
I believe this will also remove the need of aiobotocore limiting in EAGER UPGRADE . .. let see.